addons/web/static/src/legacy/scss/fields.scss

```diff
@@ -51,6 +51,16 @@
     }
 }
 
+// Invalid background
+// Avoid overlapping the bg in the input inside the list view nor on
+// the order_line because the o_invalid_cell already has the background
+:not(.o_invalid_cell) {
+    > .o_field_invalid:not(:focus-within):not(.o_form_label):not([name="order_line"]) {
```

Comment:
This selector is literally a nightmare 😅 (and I'm not considering its parent that's basically everything is not an invalid cell).
Use CSS values and adapt the bg color contextually ;)